feature/experimental/gribi/otg_tests/programming_with_reload/programming_with_reload_test.go

```diff
@@ -0,0 +1,534 @@
+}
+
+// Reload chassis
+func (args *testArgs) reloadChassis(t *testing.T) {
+
+	preRebootCompStatus := gnmi.GetAll(t, args.dut, gnmi.OC().ComponentAny().OperStatus().State())
+	t.Logf("DUT components status pre reboot: %v", preRebootCompStatus)
+
+	gnoiClient, err := args.dut.RawAPIs().BindingDUT().DialGNOI(context.Background())
+	if err != nil {
+		t.Fatalf("Error dialing gNOI: %v", err)
+	}
+	bootTimeBeforeReboot := gnmi.Get(t, args.dut, gnmi.OC().System().BootTime().State())
+	t.Logf("DUT boot time before reboot: %v", bootTimeBeforeReboot)
+
+	rebootRequest := &spb.RebootRequest{
+		Method:  spb.RebootMethod_WARM,
+		Delay:   0,
+		Message: "Reboot chassis with delay",
+		Force:   true,
+	}
+
+	t.Logf("Send reboot request: %v", rebootRequest)
+	rebootResponse, err := gnoiClient.System().Reboot(context.Background(), rebootRequest)
+	defer gnoiClient.System().CancelReboot(context.Background(), &spb.CancelRebootRequest{})
+	t.Logf("Got reboot response: %v, err: %v", rebootResponse, err)
+	if err != nil {
+		t.Fatalf("Failed to reboot chassis with unexpected err: %v", err)
+	}
+
+	startReboot := time.Now()
+	t.Logf("Wait for DUT to boot up by polling the telemetry output.")
+	for {
+		var currentTime string
+		t.Logf("Time elapsed %.2f seconds since reboot started.", time.Since(startReboot).Seconds())
+		time.Sleep(30 * time.Second)
+		if errMsg := testt.CaptureFatal(t, func(t testing.TB) {
+			currentTime = gnmi.Get(t, args.dut, gnmi.OC().System().CurrentDatetime().State())
+		}); errMsg != nil {
+			t.Logf("Got testt.CaptureFatal errMsg: %s, keep polling ...", *errMsg)
+		} else {
+			t.Logf("Device rebooted successfully with received time: %v", currentTime)
+			break
+		}
+	}
+	t.Logf("Device boot time: %.2f seconds", time.Since(startReboot).Seconds())
+
+	bootTimeAfterReboot := gnmi.Get(t, args.dut, gnmi.OC().System().BootTime().State())
+	t.Logf("DUT boot time after reboot: %v", bootTimeAfterReboot)
+	if bootTimeAfterReboot <= bootTimeBeforeReboot {
+		t.Errorf("Get boot time: got %v, want > %v", bootTimeAfterReboot, bootTimeBeforeReboot)
+	}
+
+	// starting a goroutine to attempt for gribi connection while system finishes bringup process
+	t.Logf("Starting a goroutine to attempt for gribi connection within max timeout value")
+	err_msg := make(chan error)
+	args.gribi_reconnect(t, err_msg)
+
+	t.Logf("inspect the error msg returned from goroutine for gribi connection establishment")
+	gribi_err_msg := <-err_msg
+	if gribi_err_msg != nil {
+		t.Fatalf("gribi connection fails post reboot with in max timeout value of %d", maxGribiConnectTime)
+	}
+
+	startComp := time.Now()
+	t.Logf("Wait for all the components on DUT to come up")
+
+	for {
+		postRebootCompStatus := gnmi.GetAll(t, args.dut, gnmi.OC().ComponentAny().OperStatus().State())
+		if len(preRebootCompStatus) == len(postRebootCompStatus) {
```

Comment:
This check doesn't validate that "all the components on DUT come up" as the log message on L491 indicates.

FWIW not sure if this is needed here at all - we already have separate tests that cover these scenarios and it doesnt''t seem to be the core part of the test logic here?

Reply:
Could we ltake into consideration  https://github.com/openconfig/featureprofiles/pull/2397 raised for gNOI-3.1 which aims to improve the debugability of this code-block.

Reply:
@LimeHat, I added components check as a safety net for validation, however it's not need under this scenario. I'll remove the check.